python/llm/example/Text-Generation-WebUI/README.md

```diff
@@ -0,0 +1,132 @@
+
+This tutorial provides a step-by-step guide on how to use Text-Generation-WebUI to run Hugging Face transformers-based applications on BigDL-LLM.
+
+## 1. Prepare the environment
+
+Please use a python environment management tool (we recommend using Conda) to create a python enviroment and install necessary libs.
+
+### 1.1 Install Conda
+
+For Linux users, please open a terminal and run below commands:
+
+```
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash ./Miniconda3-latest-Linux-x86_64.sh
+conda init
+```
+
+### 1.2 Create Environment
+
+Create a Python 3.9 environment with the name you choose
+
+```
+conda create -n bigdl-llm python=3.9
+```
+
+Then activate the environment
+```
+conda activate bigdl-llm
+```
+
+
+### 1.3 Install BigDL-LLM
+
+BigDL-LLM Backend WebUI has already support running applications on both CPU and GPU device, please prepare the environment according to your device and requirements.
+
+```bash
+# on cpu device
+pip install --pre --upgrade bigdl-llm[all]
+
+# on gpu device
+pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
+```
+
+### 1.4 Install other required dependencies
+
+```bash
+pip install -r requirements.txt
+```
+Note: Text-Generation-WebUI requires transformers version >= 4.36.0
+
+### 1.5 Best Known Configuration on Linux
+For optimal performance on Intel GPUs, it is recommended to set several environment variables:
+```bash
+# configure oneAPI environment variables
+source /opt/intel/oneapi/setvars.sh
+
+export USE_XETLA=OFF
+export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
+```
```

Comment:
better to separate these two. source oneapi is must and the other environment variables are optional.